migrated cypress tests to playwright

DevRish opened the pull request:
**Notes for Reviewers**

This PR fixes #10988

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.
 

<!--
Thank you for contributing to Meshery! 

Contributing Conventions:

1. Include descriptive PR titles with [<component-name>] prepended.
2. Build and test your changes before submitting a PR. 
3. Sign your commits

By following the community's contribution conventions upfront, the review process will 
be accelerated and your PR merged more quickly.
-->

aabidsofi19 commented on ui/tests/e2e/integration/settings.spec.js:
```diff
@@ -0,0 +1,71 @@
+  });
+};
+
+test.describe('Settings', () => {
```
please have proper description of what the test is

DevRish replied:
The tests related to functionality within the `Settings` page (`/settings`) are grouped inside this block, hence the name. Same is present for other pages like `User Preferences` as well. I followed the same names that were present in the previous cypress tests. 
Please suggest a naming convention / format that I should follow.

leecalcote replied:
@DevRish don't place the burden on @aabidsofi19, please. Try suggestion a naming convention. Moreover please heed his request to incorporate a code comment that describes the purpose and behavior of this test.

DevRish replied:
Done:
- Named the describe blocks containing tests for features within a particular page as `{page_name} Page Tests` 
- Named the describe blocks containing tests for features involving multiple pages as `{feature_name} Tests`
- Made the names of individual tests more descriptive
- Added comments for better understanding the flow of tests

aabidsofi19 commented on ui/tests/e2e/integration/settings.spec.js:
```diff
@@ -0,0 +1,71 @@
+};
+
+test.describe('Settings', () => {
+  test.describe('Service Meshes', () => {
```
we shouldnt have nested describe blocks .

DevRish replied:
The nested describe blocks were used for better grouping of tests. Like, using a single describe for a page we can group all tests of that page under it. But say the page has different sections, and we write multiple tests for each section. The for a `Page > Section > Test` like grouping, nested describe blocks have been used.

In this case currently there is only one test, but in the future more might get added. The nested describe blocks may help in grouping and organizing them at that time.

DevRish replied:
Done. Refactored the tests to avoid usage of nested describe blocks.

DevRish commented:
Updates:

- Fixed folder structure by bringing the tests inside existing e2e folder
- Added proper description of tests and code comments for better understanding of the flow of the tests
- Avoided the usage of nested describe blocks 

Please review @aabidsofi19 @leecalcote

aabidsofi19 commented on ui/tests/e2e/sample.spec.js:
```diff
@@ -0,0 +1,14 @@
+import { expect, test } from '@playwright/test';
```
this test case can be removed

DevRish replied:
Done

aabidsofi19 commented on ui/tests/e2e/service_mesh_lifecycle.spec.js:
```diff
@@ -0,0 +1,68 @@
+      await page.locator('[data-cy=tabServiceMeshes]').click({ force: true });
+
+      // Enter Mesh Adapter URL
+      await page.locator("[data-cy='mesh-adapter-url']").locator('input').fill('localhost:10000');
```
dont use the data-cy or id locators , try to use the semantic locators from playwright

DevRish replied:
Done

aabidsofi19 commented on ui/tests/e2e/service_mesh_lifecycle.spec.js:
```diff
@@ -0,0 +1,68 @@
+      await page.keyboard.press('Enter');
+
+      // Click 'Connect' Button
+      await page.locator('[data-cy=btnSubmitMeshAdapter]').click();
```
same

DevRish replied:
Done

aabidsofi19 commented on ui/tests/e2e/userpreference.spec.js:
```diff
@@ -0,0 +1,77 @@
+  {
+    name: 'Toggle "Meshery Catalog Content"',
+    apiURL: `${ENV.MESHERY_SERVER_URL}/api/user/prefs`,
+    locator: '[data-cy="CatalogContentPreference"]',
```
same here

DevRish replied:
Done

aabidsofi19 commented on ui/tests/e2e/userpreference.spec.js:
```diff
@@ -0,0 +1,77 @@
+
+    // Verify visibility of 'Extensions' Section
+    await expect(
+      page.locator(':nth-child(1) > .MuiFormControl-root > .MuiFormLabel-root'),
```
this will break with layout chages

DevRish replied:
Done. Changed to find the elements by their text.

vishalvivekm commented:
Thank you @DevRish 
Let's discuss this in Meshery Dev meeting starting in 20 minutes.
Meeting minutes: https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit?pli=1#
Meeting link: https://meet.layer5.io/meshery

DevRish commented:
Updates:

- Refactored usage of `data-cy` to playwright's semantic locators
- Removed the sample tests
- Improved API interaction interceptions

Please review @aabidsofi19
